### PR TITLE
Improve e2e tests with bashunit 0.17

### PR DESCRIPTION
**Pull request by Chemaclass**
Closed in favor of https://github.com/phpstan/phpstan-src/pull/3533

### CODE REVIEW

**Comment by staabm** [comment]
Nice 👍

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -160,49 +160,49 @@ jobs:
               cd e2e/trait-caching
               ../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/
               patch -b data/TraitOne.php < TraitOne.patch
-              OUTPUT=$(../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/ || true)
+              OUTPUT=$(../bashunit -a exit_code "1" "../../bin/phpstan analyze --no-progress --level 8 --error-format raw data/")
               echo "$OUTPUT"
-              ../bashunit -a line_count 1 "$OUTPUT"
+              ../bashunit -a line_count 2 "$OUTPUT"
               ../bashunit -a contains 'Method TraitsCachingIssue\TestClassUsingTrait::doBar() should return stdClass but returns Exception.' "$OUTPUT"
```
Comment (Chemaclass) [inline .github/workflows/e2e-tests.yml:166]: I wonder if we want to assert also  this new line that just appear? Or is it OK like this?
Same for lines 173 & 182

**Comment by staabm** [comment]
you should use rebase and get rid of this merge commit

**Comment by Chemaclass** [comment]
Closed in favor of https://github.com/phpstan/phpstan-src/pull/3533

**Pull request closed without merging**